File: src/MaksIT.WebUI/src/models/identity/user/search/SearchUserResponse.ts
import { ResponseModelBase } from '../../../ResponseModelBase'

/** Matches <c>MaksIT.CertsUI.Models.Identity.User.Search.SearchUserResponse</c>. */
export interface SearchUserResponse extends ResponseModelBase {
  id: string
  username: string
  email?: string | null
  mobileNumber?: string | null
  isActive: boolean

  twoFactorEnabled?: boolean
  recoveryCodesLeft?: number | null

  createdAt: string
  lastLogin?: string | null

  isGlobalAdmin?: boolean
  /** Present when the backend scopes rows by organization (used for ACL checks). */
  organizationId?: string | null
}

File: src/MaksIT.CertsUI/Models/Identity/User/Search/SearchUserResponse.cs
using MaksIT.Core.Abstractions.Webapi;


namespace MaksIT.CertsUI.Models.Identity.User.Search;

public class SearchUserResponse : ResponseModelBase {
  public required Guid Id { get; set; }

  #region Master data
  public required string Username { get; set; }
  public string? Email { get; set; }
  public string? MobileNumber { get; set; }
  public bool IsActive { get; set; }
  #endregion

  #region Two-factor authentication
  public bool TwoFactorEnabled { get; set; }
  public int? RecoveryCodesLeft { get; set; }
  #endregion

  public DateTime CreatedAt { get; set; }
  public DateTime? LastLogin { get; set; }

  #region Authorization
  public bool IsGlobalAdmin { get; set; }
  #endregion
}
